Yoast/wordpress-seo
# Rewritten robots presenter/presentation code, added robots presentati…

Rewritten robots presenter/presentation code, added robots presentation code for post type archives

## Summary


This PR can be summarized in the following changelog entry:

* N/A

## Relevant technical choices:

* Added traits for presentations unit tests to reduce boilerplate.
* 

## Test instructions

This PR can be tested by following these steps:

* Install Yoast test helper.
* Activate books and movies custom post types.
* In `Search appearance` > `Content Types` set `Show the archive for Books in search results?` to `No`.
* In `Search appearance` > `Content Types` set `Show the archive for Movies in search results?` to `No`.
* Go to the Books archive (yourdomain`/my-books`) and check that the `robot` metatag is `noindex,nofollow`.
* Go to the Movies archive (yourdomain`/yoast-test-movies`) and check that the `robot` metatag is `noindex,nofollow`.
* In `Search appearance` > `Content Types` set `Show the archive for Books in search results?` to `Yes`.
* In `Search appearance` > `Content Types` set `Show the archive for Movies in search results?` to `Yes`.
* Go to the Books archive (yourdomain`/my-books`) and check that the `robot` metatag is not present because `index,follow` is default behaviour.
* Go to the Movies archive (yourdomain`/yoast-test-movies`) and check that the `robot` metatag is not present because `index,follow` is default behaviour.

## UI changes

## Documentation

## Quality assurance
